src/controllers/source-statuses.controller.ts

```diff
@@ -1,43 +1,84 @@
+import { NotUniqueError } from '../errors/not-unique.error';
+import { RemovalRestrictedError } from '../errors/removal-restricted.error';
 
 export class SourceStatusesController {
-  static test = async (req: Request, res: Response) => {
-    res.send('Test response');
-  };
-
-  static async getAll(req: Request, res: Response) {
-    const source_statuses = await SourceStatus.find();
-    res.send(source_statuses);
+  static async getSourceStatuses(req: Request, res: Response): Promise<Response<SourceStatusesService>> {
+    try {
+      const source_statuses = await SourceStatusesService.getAll();
+      return res.json({ data: source_statuses });
+    } catch (e) {
+      return res.status(500).json({ message: e.message });
+    }
   }
 
-  static async getOneById(req: Request, res: Response) {
-    const id: string = req.params.id;
+  static async getSourceStatusById(req: Request, res: Response): Promise<Response<SourceStatusesService>> {
+    try {
+      const source_status = await SourceStatusesService.getOneById(Number(req.params.id));
+      return res.json({ data: source_status });
+    } catch (e) {
+      if (e instanceof NotFoundError) {
+        res.status(404).json({ message: e.message });
+      }
+      return res.status(500).json({ message: e.message });
+    }
+  }
 
+  static async createSourceStatus(req: Request, res: Response): Promise<Response<SourceStatusesService>> {
     try {
-      const source_status = await SourceStatus.findOneOrFail(id, {
-        select: ['id', 'name', 'description', 'sources'],
-      });
-      res.send(source_status);
-    } catch (error) {
-      res.status(404).send('Source status not found');
+      const { name, description } = req.body;
+      const new_source_status = await SourceStatusesService.addOne(name, description);
+      return res.json({ data: new_source_status });
+    } catch (e) {
+      if (e instanceof NotFoundError) {
+        return res.status(404).json({ message: e.message });
+      } else if (e instanceof NotUniqueError) {
+        return res.status(400).json({ message: e.message });
+      }
+      return res.status(500).json({ message: e.message });
     }
   }
 
-  static async add(req: Request, res: Response) {
-    const { name, description } = req.body;
+  static async updateSourceStatus(req: Request, res: Response): Promise<Response<SourceStatus>> {
+    try {
+      const source_status: {
+        source_status_id: number;
+        name: string;
+        description: string;
+      } = req.body;
 
-    const source_status = new SourceStatus();
-    source_status.name = name;
-    source_status.description = description;
+      source_status.source_status_id = Number(req.params.id);
 
-    try {
-      await source_status.save();
+      const updated_source_status = await SourceStatusesService.updateOne(
+        source_status.source_status_id,
+        source_status.name,
+        source_status.description
```

Comment:
Because validation makes sure the body has valid data, we could use data from the body as service method parameters directly and get rid of `source_status` object entirely. Feel free to do this in all methods of all controllers.